src/main/java/io/anserini/analysis/vectors/fw/EncodeAndQuantizeFilter.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Anserini: A Lucene toolkit for replicable information retrieval research
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.anserini.analysis.vectors.fw;
+
+import org.apache.lucene.analysis.TokenFilter;
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.tokenattributes.CharTermAttribute;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+import static io.anserini.analysis.vectors.fw.FakeWordsEncoderAnalyzer.REMOVE_IT;
+
+/**
+ * {@link TokenFilter} that encodes a real valued token into a stream of "fake word" tokens proportional to
+ * the corresponding quantized input (real) value.
+ *
+ * Note: if quantization leads to zero, a "to-be-removed" token is added (and eventually taken care of by a
+ * {@link org.apache.lucene.analysis.StopFilter} in the downstream text analysis chain.
+ */
+public final class EncodeAndQuantizeFilter extends TokenFilter {
```

Comment:
Would it make sense to rename to `FakeWordsEncodeAndQuantizeFilter` so that it is clear what it does, without needing to look at the package it's in?

Reply:
+1